astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1142,6 +1154,7 @@ def postinit(
         body: list[NodeNG],
         decorators: node_classes.Decorators | None = None,
         returns=None,
+        type_params: list[nodes.TypeVar] | None = None,
```

Comment:
The constructor has this as always being a list. Do we want to make this always a list?

Reply:
We'd end up with `dangerous-default-value` if we provide a default list. We could leave as is, or remove any default. Thoughts?

I should make it keyword-only in any case, that's not nice to alter the order of positional arguments.

Reply:
I can check for None and provide `[]` instead to the instance attribute if the method arg is None.

Reply:
I think we'll have to give `None` but just check if `None` and if so don't assign.
I'm wondering if there is ever a need to distinguish between `None` and `[]`, but I think `None` should only be used if we can't know. That doesn't seem to be the case here since we aren't using `inspect`.

Reply:
Ah you commented just before me, but yeah let's do that!